File: chapter 1/1.3.py
# 1.3
# URLify: Write a method to replace all spaces in a string with '%20'. You may assume that the string
# has sufficient space at the end to hold the additional characters, and that you are given the "true"
# length of the string. (Note: If implementing in Java, please use a character array so that you can
# perform this operation in place.) 

def URLify (string):
    if ' ' not in string:
        return string
    else:
        list = string.split(" ")
        newString = ""
        for i in range(0,len(list)-1):
            newString += list[i]
            newString += "%20"
        newString += list[len(list)-1]
        return newString

print(URLify("Mr Jones hello"))    

1.4
Palindrome Permutation: Given a string, write a function to check if it is a permutation of a palindrome. A palindrome is a word or phrase that is the same forwards and backwards. A permutation
is a rearrangement of letters. The palindrome does not need to be limited to just dictionary words. 


racecarr ayya \{r:2, a:2, c:2, e:1} {r:true a:truec:true e:false}
def palidnromePermutation (oldStr):
    dictionary = {}
    str = oldStr.lower()

    for i in range(len(str)):
        if str[i] not in dictionary and str[i] != " "  :
            dictionary[str[i]] = False
        elif str[i] in dictionary:
            dictionary[str[i]] = not dictionary[str[i]]
        
    print(dictionary)
    f = 0
    for key, value in dictionary.items():
        if value == False:
            f = f + 1

        if f > 1: 
           return False

    return True
print(palidnromePermutation("Taco cat"))

File: chapter 1/js/1.3.js
// URLify: Write a method to replace all spaces in a string with '%20'. You may assume that the string
// has sufficient space at the end to hold the additional characters, and that you are given the "true"
// length of the string. (Note: If implementing in Java, please use a character array so that you can
// perform this operation in place.) 

function uRLify(str){
    if(str.length == 0){
        return str;
    }
    let strArr=str.split(" ");
    console.log(strArr)
    let newStr = "";
    for (let i=0; i<strArr.length-1; i++){
        newStr = newStr + strArr[i] + "%20";
    }
    newStr = newStr + strArr[strArr.length-1];
    return newStr;
}

// function uRLify(str){
//     if(str.length == 0){
//         return str;
//     }
//     let count = 0;
//     for (let i=0; i<str.length; i++){
//         if(str[i] == " "){
//             count ++;
//         }
//     }
//     count = count * 2;
//     for (let i=str.length-1; i>=0; i--){
//         if(str[i]!= " "){
//             str[i+count] = str[i]
//         }
//         if(str[i] == " "){
//             str[i] = "%20"
//         }
//         console.log("inside for loop: " + str)
//     }
//     return str;
// }

// console.log(uRLify("st ri ng"))
